tower/src/retry/backoff.rs

```diff
@@ -0,0 +1,263 @@
+//! This module contains generic [backoff] utlities to be used with the retry
+//! layer.
+//!
+//! The [`Backoff`] trait is a generic way to represent backoffs that can use
+//! any timer type.
+//!
+//! [`ExponentialBackoffMaker`] implements the maker type for  
+//! [`ExponentialBackoff`] which implements the [`Backoff`] trait and provides
+//! a batteries included exponential backoff and jitter strategy.
+//!
+//! [backoff]: https://en.wikipedia.org/wiki/Exponential_backoff
+
+use std::fmt::Display;
+use std::future::Future;
+use std::time::Duration;
+use tokio::time;
+
+use crate::util::rng::{HasherRng, Rng};
+
+/// Trait used to construct [`Backoff`] trait implementors.
+pub trait MakeBackoff {
```

Comment:
I don't know why, something about having a second trait feels less ideal, but I admit I can't think of a better way (other than making Backoff return a new Backoff each time, but that makes it no longer object-safe...). :shrug:

Reply:
Yeah, I agree, I have the same feeling but this feels cleaner than any other approach I could find.